core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala

```diff
@@ -1036,6 +1036,28 @@ class GroupCoordinatorTest {
     assertEquals(Errors.NONE, syncGroupWithOldMemberIdResult.error)
   }
 
+ @Test
+ def staticMemberJoinStorageWithUnknownPersistenceFailure(): Unit = {
+   val rebalanceResult = staticMembersJoinAndRebalance(leaderInstanceId, followerInstanceId)
+   val joinGroupResult = staticJoinGroupWithPersistence(groupId, JoinGroupRequest.UNKNOWN_MEMBER_ID, followerInstanceId, protocolType, protocolSuperset, clockAdvance = 1, 2 * DefaultSessionTimeout, 2 *DefaultSessionTimeout, appendRecordError = Errors.MESSAGE_TOO_LARGE)
+   checkJoinGroupResult(joinGroupResult,
+     Errors.UNKNOWN_SERVER_ERROR,
+     rebalanceResult.generation,
+     Set.empty,
+     Stable,
+     Some(protocolType))
+   assertNotEquals(None, groupCoordinator.groupManager.getGroup(groupId))
+   val group = groupCoordinator.groupManager.getGroup(groupId).get
+   val it = group.allMemberMetadata.iterator
+   while (it.hasNext) {
+     val member = it.next()
+     assertNotEquals(member.sessionTimeoutMs, 2 * DefaultSessionTimeout)
+     assertEquals(member.sessionTimeoutMs, DefaultSessionTimeout)
+     assertNotEquals(member.rebalanceTimeoutMs, 2 * DefaultRebalanceTimeout)
+     assertEquals(member.rebalanceTimeoutMs, DefaultRebalanceTimeout)
+   }
+ }
+
```

Comment:
Could we also add a unit test which verifies that the happy case? I mean, when persisting the change succeeds.

Reply:
I Have add a test `staticMemberJoinStorageWithUnknownPersistenceSuccess`